samcli/lib/bootstrap/bootstrap.py

```diff
@@ -19,9 +19,17 @@
 
 
 def manage_stack(profile, region):
-    session = boto3.Session(profile_name=profile if profile else None)
-    cloudformation_client = session.client("cloudformation", config=Config(region_name=region if region else None))
-
+    try:
+        session = boto3.Session(profile_name=profile if profile else None)
+        cloudformation_client = session.client("cloudformation", config=Config(region_name=region if region else None))
+    except NoCredentialsError:
+        raise UserException(
```

Comment:
we need to change userexception to a more specific exception.

merge after adding unit tests